server/src/main/java/org/candlepin/resource/ActivationKeyResource.java

```diff
@@ -321,32 +209,25 @@ public ActivationKeyDTO removeProductIdFromKey(
     @Produces(MediaType.APPLICATION_JSON)
     public void deleteActivationKey(
         @PathParam("activation_key_id")
-        @Verify(ActivationKey.class) String activationKeyId) {
-        ActivationKey key = this.fetchActivationKey(activationKeyId);
-
+        @Verify(ActivationKey.class)
+        @NotEmpty String activationKeyId) {
         log.debug("Deleting activation key: {}", activationKeyId);
-
-        activationKeyCurator.delete(key);
-    }
-
-    private Pool findPool(String poolId) {
-        Pool pool = poolManager.get(poolId);
-
-        if (pool == null) {
-            throw new BadRequestException(i18n.tr("Pool with id {0} could not be found.", poolId));
-        }
-
-        return pool;
+        this.activationKeyController.deleteActivationKey(activationKeyId);
     }
 
-    private Product confirmProduct(Owner o, String prodId) {
-        Product prod = this.ownerProductCurator.getProductById(o, prodId);
+    private ActivationKey toActivationKey(ActivationKeyDTO update) {
```

Comment:
Now that we have agreed to introduce automatic reverse-translation, this should be replaced by a new Translator class and call to that translator framework.